research/conditional/utils/conditional_trainer.py

```diff
@@ -455,8 +487,13 @@ def _save_weights(self, step):
                 self.save_weights_path,
                 self.rank,
                 step,
+                self.logger,
             )
 
+            return True
+        else:
+            return False
```

Comment:
I would split decision if we should stop xperitment from saving weights

Reply:
I have changed that function name, now stopping experiment propagate from repeater_rerun functionality that additionally saves training and model params.